src/OrchardCore.Modules/OrchardCore.Users/ViewModels/RegisterExternalLoginViewModel.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace OrchardCore.Users.ViewModels
+{
+    public class RegisterExternalLoginViewModel : IValidatableObject
+    {
+        public bool NoUsername { get; set; }
+        public bool NoEmail { get; set; }
+        public bool NoPassword { get; set; }
+
+        public string UserName { get; set; }
+
+        [EmailAddress]
+        public string Email { get; set; }
+
+        [DataType(DataType.Password)]
+        public string Password { get; set; }
+
+        [DataType(DataType.Password)]
+        public string ConfirmPassword { get; set; }
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            if (string.IsNullOrWhiteSpace(Email) && !NoEmail)
+            {
+                yield return new ValidationResult("Email is required!", new[] { "Email" });
```

Comment:
Shouldn't it be string-localized? Have you seen this pattern somewhere else that we should fix too?

Reply:
Opened an issue #4619